src/prefect/core/task.py

```diff
@@ -504,39 +504,55 @@ def set_dependencies(
             mapped=mapped,
         )
 
-    def set_upstream(self, task: object, flow: "Flow" = None) -> None:
+    def set_upstream(
+        self, task: object, flow: "Flow" = None, key: str = None, mapped: bool = False
+    ) -> None:
         """
         Sets the provided task as an upstream dependency of this task.
 
-        Equivalent to: `self.set_dependencies(upstream_tasks=[task])`
-
         Args:
             - task (object): A task or object that will be converted to a task that will be set
                 as a upstream dependency of this task.
             - flow (Flow, optional): The flow to set dependencies on, defaults to the current
                 flow in context if no flow is specified
+            - key (str, optional): The key to be set for the new edge; this is
```

Comment:
I can't do a multiline suggest but I think the wording is a bit confusing (in particular telling which is the upstream and which is the downstream):

```
- key (str, optional): The key to be set for the new edge. The result of the 
upstream task will be passed to this task's `run()` method under this 
keyword argument.
```

Reply:
Updated here and elsewhere.